scripts/tests/sanitycheck/test_reporting_testsuite.py

```diff
@@ -0,0 +1,151 @@
+#!/usr/bin/env python3
+# Copyright (c) 2020 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+
+'''
+This test file contains testcases for reporting functionality of Testsuite class of sanitycheck
+'''
+import sys
+import os
+import xml.etree.ElementTree as ET
+import pytest
+import pandas as pd
+from mock import MagicMock
+
+ZEPHYR_BASE = os.getenv("ZEPHYR_BASE")
+sys.path.insert(0, os.path.join(ZEPHYR_BASE, "scripts/sanity_chk"))
+
+
+def test_discard_report(class_testsuite, platforms_list, all_testcases_dict, caplog, tmpdir):
+    """ Testing discard_report function of Testsuite class in sanitycheck
+	Test 1: Check if apply_filters function has been run before running
+	discard_report
+	Test 2: Test if the generated report is not empty
+	Test 3: Test if the gerenrated report contains the expected columns"""
+    class_testsuite.platforms = platforms_list
+    class_testsuite.testcases = all_testcases_dict
+    filename = tmpdir.mkdir("test_discard").join("discard_report.csv")
```

Comment:
what is the reason to create directories for all reports if you have only file with a name that does not collide with other files ?

Reply:
This is the temporary direction and the purpose to use the temporary directory is that these directories get deleted after execution of the testcase. This is the fixture provided by pytest.